docs/how-to/backend-for-frontend.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Backend For Frontend
+---
+
+# Backend For Frontend
+
+While React Router can serve as your fullstack application, it also fits perfectly into the "Backend for Frontend" architecture.
+
+The BFF strategy employs a web server with a job scoped to serving the frontend web app and connecting it to the services it needs: your database, mailer, job queues, existing backend APIs (REST, GraphQL), etc. Instead of your UI integrating directly from the browser to these services, it connects to the BFF, and the BFF connects to your services.
+
+Mature apps already have a lot of backend application code in Ruby, Elixir, PHP, etc., and there's no reason to justify migrating it all to a server-side JavaScript runtime just to get the benefits of React Router. Instead, you can use your React Router app as a backend for your frontend.
+
+You can use `fetch` right from your loaders and actions to your backend.
+
+```tsx lines=[10,16,20]
+import type { Route } from "./+types/some-route";
+import escapeHtml from "escape-html";
+
+export async function loader({
+  request,
+}: Route.LoaderArgs) {
```

Comment:
We can simplify this a bit. I'm not totally sure why the original example had the types, but since this isn't using the `request` and more talking about a concept, let's simplify it a bit

```suggestion
```tsx lines=[7,13,17]
import escapeHtml from "escape-html";

export async function loader() {
```